pandas/tests/groupby/test_groupby.py

```diff
@@ -1698,3 +1698,19 @@ def test_groupby_agg_ohlc_non_first():
     result = df.groupby(pd.Grouper(freq='D')).agg(['sum', 'ohlc'])
 
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_agg_coercing_bools():
+    # issue 14873
+    dat = pd.DataFrame(
+        {'a': [1, 1, 2, 2], 'b': [0, 1, 2, 3], 'c': [None, None, 1, 1]})
+    gp = dat.groupby('a')
+    case_one_res = gp['b'].aggregate(lambda x: (x != 0).all())
```

Comment:
can you write this a 2 successve cases, using
result = 
expected = 
assert_series_equal(result, expected)

for each one (can be in the same test)

Reply:
@jreback made the suggested changes.